### ISSUE TITLE
Problems verifying types when supertypes are not implemented

### ISSUE THREAD

**Issue by cmgrote** [odpi/egeria#2965]
When implementing a connector that implements only the "leaf" types (eg. `RelationalColumn`) and not the supertypes directly (eg. `SchemaElement`) there are problems doing the type verification that appear in the CTS (and perhaps elsewhere)?

For example, in 1.6 we added the `isDeprecated` property at the `SchemaElement` level.  In a connector like the IGC connector, the `SchemaElement` level is not directly implemented or mapped, but adding this `isDeprecated` property directly to the `RelationalColumn` implementation is also not sufficient -- because the supertype `SchemaElement` is never patched with this additional property (because `SchemaElement` is from the beginning listed by the connector as not implemented -- so no patch is attempted to be applied to it).

This could also be related to the `updateTypeDef` method being optional -- if not implemented, patching of type defs cannot be done, in which case the CTS should (presumably) operate based on the original (unpatched) typedefs rather than the latest version(s) of a typedef?  But this does not seem to be the case, either: the CTS seems to be trying to test the latest version of types in certain optional profiles, causing a `PropertyErrorException` to be thrown by the parameter validation before ever reaching the `FunctionNotSupportedException` that would otherwise be thrown for an optional method:

- repository-reference-copy-classification-lifecycle
- repository-entity-lifecycle
- repository-classification-lifecycle
- repository-relationship-lifecycle
- repository-entity-reidentify
- repository-relationship-reidentify
- repository-entity-retype

Attached are example negative results for each of these optional profiles, when `updateTypeDef` is not implemented and the supertypes are also not directly implemented: 
[exampleNegativeResults.json.zip](https://github.com/odpi/egeria/files/4521966/exampleNegativeResults.json.zip)

(The full results from which these were extracted can be seen at: https://github.com/odpi/egeria-connector-ibm-information-server/blob/master/cts/results/1.6/11.5.0.2sp5/openmetadata.conformance.testlab.results)

**Comment by cmgrote** [comment]
The above is the behaviour observed in both 1.6 and candidate 1.7 release; however, the problem appears to be worse in the candidate 1.7 release as it also causes the "Support for consistent types" CTS check to fail with results like the following, which has the knock-on impact of causing the entire mandatory metadata sharing profile to be marked as `NOT_CONFORMANT`:

```json
{
  "id": 5,
  "name": "Support for consistent types",
  "description": "The technology under test supports entity, relationship and classification types that link together.",
  "documentationURL": "https://egeria.odpi.org/open-metadata-conformance-suite/docs/repository-workbench/profiles/metadata-sharing/consistent-types",
  "conformanceStatus": "NOT_CONFORMANT",
  "negativeTestEvidence": [
    {
      "profileId": 0,
      "requirementId": 5,
      "testCaseId": "repository-entity-property-search-RelationalColumn",
      "testCaseName": "Repository entity property search test case",
      "testCaseDescriptionURL": "https://egeria.odpi.org/open-metadata-conformance-suite/docs/repository-workbench/test-cases/repository-entity-property-search-test-case.md",
      "testEvidenceType": "UNSUCCESSFUL_ASSERTION",
      "assertionMessage": "RelationalColumn entity type matches the known type from the repository helper."
    },
    {
      "profileId": 0,
      "requirementId": 5,
      "testCaseId": "repository-entity-reference-copy-lifecycle-RelationalColumn",
      "testCaseName": "Repository entity reference copy lifecycle test case",
      "testCaseDescriptionURL": "https://egeria.odpi.org/open-metadata-conformance-suite/docs/repository-workbench/test-cases/repository-entity-reference-copy-lifecycle-test-case.md",
      "testEvidenceType": "UNSUCCESSFUL_ASSERTION",
      "assertionMessage": "RelationalColumn entity type definition matches known type  "
    }
  ]
}
```

Not sure what changed between 1.6 and 1.7, but this particular check in 1.6 passed and now fails under 1.7.

**Comment by planetf1** [comment]
Any updates on whether we need this for 1.7? There was  a minor change to the repository validator in PR #2970 (about to get into 1.7) but it doesn't look related. 

I am a little concerned we have a regression in documented compliance, though from the above it maybe that type patching is needed as mandatory in the profile? Clearly the impact is limited to only repo connectors so small currently - and on that bases a readme entry may suffice?

**Comment by mandy-chessell** [comment]
I think this actually needs to be fixed in the OMRS runtime not the CTS.  We need to make sure property validation works for these types of repositories.  I think we should just fix in master unless it is something that used to work in 1.6 and we have changed something to break the repository helper?

**Comment by cmgrote** [comment]
> I think we should just fix in master unless it is something that used to work in 1.6 and we have changed something to break the repository helper?

Hard to say -- at least in 1.6 it was possible to get a full conformance result for the `"Support for consistent types"` test; in 1.7 it is not.  Not clear what caused this though: whether a change to repository helpers, types, or CTS...

Should we just accept that this particular test will fail for the external connectors against the 1.7 release?

**Comment by mandy-chessell** [comment]
I am confused about the problem being reported here.  When we discussed it, I thought the problem was that the base classes for the metadata collection were rejecting the supertype's new property because they were only looking at the active types (ie the types that the repository connector indicated they supported via the `verifyTypeDef` method.

However I have just checked the code and it works from knownTypeDefs so it should be fine.  This suggests the problem is in the specific implementation of the connector rather than the OMRS code?
